File: caesar_cipher/js/caesarCipher.js
exports.caesarCipher = function(string, shift_amount) {
let answerString=[]
let newArr=string.split("")
for (let i=0; i<newArr.length; i++){
    let is_uppercase =false
    let x=newArr[i]
    if (newArr[i].replace(/[^a-zA-Z]+/,"DID")!=="DID")
        {
        // console.log(x)
        if (newArr[i]!==newArr[i].toLowerCase()){
            is_uppercase=true
        }
        x=x.toLowerCase()
        x=x.charCodeAt()
        x+=shift_amount
        if (x<97){
            x=123+(x-97)}
        else if (x>122){
            x=96+(x-122)}
        x=String.fromCharCode(x)
        // console.log(x, "THIS IS CHANGED")
        if (is_uppercase===true){
            x=x.toUpperCase()
        }
        // console.log(x, "final")
        }
    answerString.push(x)
    // console.log(newArr[i])
}
return answerString.join("")
};
// console.log((caesarCipher("ABC DEFGHIJKLMNOPQURSTUVWXYZ", 5))) 
// "Wjt! Rcvo v nomdib!"

File: caesar_cipher/python/caesar_cipher.py
from operator import is_


def caesar_cipher(string, shift_amount):
    answer_string=[]
    for count, x in enumerate(string):
        if x.isalpha():
            is_uppercase=False
            if x!=x.lower():
                is_uppercase=True
            # print(x, "startingvalue")
            x=ord(x.lower())
            # print(x)
            x+=shift_amount
            # print(x)
            if x<97:
                x=123+(x-97)
            elif x>122:
                x=96+(x-122)
            x=chr(x)
            if is_uppercase==True:
                x=x.upper()
            # print(x, "endingvalue")
        answer_string.append(x)
    return "".join(answer_string)

# if ord(x)>96 and ord(x)<123:
# print(chr(122))

print(caesar_cipher("ABC DEFGHIJKLMNOPQURSTUVWXYZ", 5)) #"Wjt! Rcvo v nomdib!"
